enterprise/cmd/frontend/internal/executorqueue/handler/handler.go

```diff
@@ -23,12 +23,15 @@ type QueueOptions struct {
 	// RecordTransformer is a required hook for each registered queue that transforms a generic
 	// record from that queue into the job to be given to an executor.
 	RecordTransformer func(ctx context.Context, record workerutil.Record) (apiclient.Job, error)
+
+	// FetchCanceled is an optional hook that can be provided to support cancelation.
+	// If it is set, it will be invoked periodically and should return the IDs to be
+	// canceled for the given executor.
+	FetchCanceled func(ctx context.Context, executorName string) (canceledIDs []int, err error)
```

Comment:
```suggestion
	CanceledRecordFetcher func(ctx context.Context, executorName string) (canceledIDs []int, err error)
```

To be consistent with the naming